web/src/dropdown_list_widget.js

```diff
@@ -201,6 +201,14 @@ export class DropdownListWidget {
                     break;
                 }
             }
+
+            if (keydown_util.is_enter_event(e) && e.target === $search_input[0]) {
+                e.stopPropagation();
+                e.preventDefault();
+                // Select the first option from the menu on pressing
+                // "Enter" when focus is on the search input.
+                dropdown_elements().first().trigger("click");
```

Comment:
Did not add this in the switch statements since we check enter event using `is_enter_event` at other places as well and not using `e.key`. Also, added this code here instead of `register_event_handlers` since we can already use `dropdown_elements` and `$search_input` variable defined here. Let me know, if we instead want this in `register_event_handlers` only.

Reply:
Yeah, this looks great.